src/link.cpp

```diff
@@ -488,18 +493,28 @@ static void construct_linker_job_coff(LinkJob *lj) {
             if (lj->codegen->zig_target.env_type == ZigLLVM_GNU) {
                 Buf *arg = buf_sprintf("-l%s", buf_ptr(link_lib->name));
                 lj->args.append(buf_ptr(arg));
-            }
-            else {
+            } else {
                 lj->args.append(buf_ptr(link_lib->name));
             }
         } else {
             buf_resize(def_contents, 0);
             buf_appendf(def_contents, "LIBRARY %s\nEXPORTS\n", buf_ptr(link_lib->name));
+#ifdef ZIG_OS_WINDOWS
+            Buf* dll_name = buf_create_from_buf(link_lib->name);
+            buf_append_str(dll_name, ".dll");
+            HMODULE hmod = GetModuleHandle(buf_ptr(dll_name));
```

Comment:
From MSDN GetModuleHandle:
> Retrieves a module handle for the specified module. The module must have been loaded by the calling process.

Based on this, I think we have to use LoadLibrary before GetModuleHandle. And LoadLibrary runs the code in the DLL, which can crash the compiler. So we should do it in a child process, and communicate back to zig compiler the results. If we're going to spawn a child process, we might as well write that child process in zig and build it lazily just like compiler_rt.o and builtin.o.